armi/utils/tests/test_plotting.py

```diff
@@ -88,6 +89,26 @@ def test_plotBlockFlux(self):
             os.remove("bList2.txt")  # secondarily created during the call.
             os.remove("bList2.png")  # created during the call.
 
+    def test_plotHexBlock(self):
+        o, r = test_reactors.loadTestReactor()
+        first_fuel_block = r.core.getFirstBlock(Flags.FUEL)
+        first_fuel_block.autoCreateSpatialGrids()
+        plotting.plotBlockDiagram(first_fuel_block, "blockDiagram23.svg", True)
+        self._checkExists("blockDiagram23.svg")
+
+    def test_plotCartesianBlock(self):
+        from armi import settings
+        from armi.reactor import blueprints, reactors
+
+        cs = settings.Settings(
+            os.path.join(TEST_ROOT, "tutorials", "c5g7-settings.yaml")
+        )
+        blueprint = blueprints.loadFromCs(cs)
+        r = reactors.factory(cs, blueprint)
+        for b in r.core.getBlocks():
+            plotting.plotBlockDiagram(b, "blockDiagram25.svg", True)
```

Comment:
Plotting all of the blocks in the core is pretty heavy handed. Since you have the blueprints, try iterating the blockDesigns on those, constructing the blocks, then plotting those.